## Practice Exercise 2

In this assignment, you will try to find some interesting insights into a few movies released between 1916 and 2016, using Python. You will have to import the movie dataset, write Python code to explore the data, gain insights into the movies, actors, directors, and collections, and submit the code.

#### Some tips before starting the assignment

1. Identify the task to be performed correctly, and only then proceed to write the required code. Don’t perform any incorrect analysis or look for information that isn’t required for the assignment.
2. In some cases, the variable names have already been assigned, and you just need to write code against them. In other cases, the names to be given are mentioned in the instructions. We strongly advise you to use the mentioned names only.
3. Always keep inspecting your data frame after you have performed a particular set of operations.
4. There are some checkpoints given in the IPython notebook provided. They're just useful pieces of information you can use to check if the result you have obtained after performing a particular task is correct or not.
5. Note that you will be asked to refer to documentation for solving some of the questions. That is done on purpose for you to learn new commands and also how to use the documentation.

In [1]:
# Import the numpy and pandas packages

import numpy as np
import pandas as pd

### Task 1: Reading and Inspection

**Subtask 1.1: Import and read**

Import and read the movie database. Store it in a variable called `movies`.

In [2]:
# Write your code for importing the csv file here
movies = pd.read_csv('Movies.csv')
movies

color      director_name  num_critic_for_reviews  duration  \
0     Color      James Cameron                   723.0     178.0   
1     Color     Gore Verbinski                   302.0     169.0   
2     Color         Sam Mendes                   602.0     148.0   
3     Color  Christopher Nolan                   813.0     164.0   
4     Color     Andrew Stanton                   462.0     132.0   
...     ...                ...                     ...       ...   
3848  Color      Shane Carruth                   143.0      77.0   
3849  Color   Neill Dela Llana                    35.0      80.0   
3850  Color   Robert Rodriguez                    56.0      81.0   
3851  Color       Edward Burns                    14.0      95.0   
3852  Color           Jon Gunn                    43.0      90.0   

      director_facebook_likes  actor_3_facebook_likes        actor_2_name  \
0                         0.0                   855.0    Joel David Moore   
1                       563.0                  1000.0       Orlando Bloom   
2                         0.0                   161.0        Rory Kinnear   
3                     22000.0                 23000.0      Christian Bale   
4                       475.0                   530.0     Samantha Morton   
...                       ...                     ...                 ...   
3848                    291.0                     8.0      David Sullivan   
3849                      0.0                     0.0     Edgar Tancangco   
3850                      0.0                     6.0     Peter Marquardt   
3851                      0.0                   133.0  Caitlin FitzGerald   
3852                     16.0                    16.0    Brian Herzlinger   

      actor_1_facebook_likes        gross  \
0                     1000.0  760505847.0   
1                    40000.0  309404152.0   
2                    11000.0  200074175.0   
3                    27000.0  448130642.0   
4                      640.0   73058679.0   
...                      ...          ...   
3848                   291.0     424760.0   
3849                     0.0      70071.0   
3850                   121.0    2040920.0   
3851                   296.0       4584.0   
3852                    86.0      85222.0   

                                   genres  ... num_user_for_reviews language  \
0         Action|Adventure|Fantasy|Sci-Fi  ...               3054.0  English   
1                Action|Adventure|Fantasy  ...               1238.0  English   
2               Action|Adventure|Thriller  ...                994.0  English   
3                         Action|Thriller  ...               2701.0  English   
4                 Action|Adventure|Sci-Fi  ...                738.0  English   
...                                   ...  ...                  ...      ...   
3848                Drama|Sci-Fi|Thriller  ...                371.0  English   
3849                             Thriller  ...                 35.0  English   
3850  Action|Crime|Drama|Romance|Thriller  ...                130.0  Spanish   
3851                         Comedy|Drama  ...                 14.0  English   
3852                          Documentary  ...                 84.0  English   

          country  content_rating       budget  title_year  \
0             USA           PG-13  237000000.0      2009.0   
1             USA           PG-13  300000000.0      2007.0   
2              UK           PG-13  245000000.0      2015.0   
3             USA           PG-13  250000000.0      2012.0   
4             USA           PG-13  263700000.0      2012.0   
...           ...             ...          ...         ...   
3848          USA           PG-13       7000.0      2004.0   
3849  Philippines       Not Rated       7000.0      2005.0   
3850          USA               R       7000.0      1992.0   
3851          USA       Not Rated       9000.0      2011.0   
3852          USA              PG       1100.0      2004.0   

     actor_2_face

**Subtask 1.2: Inspect the dataframe**

Inspect the dataframe's columns, shapes, variable types etc.

In [3]:
# Write your code for inspection here
movies.shape



(3853, 28)

#### <font color='red'>Question 1: How many rows and columns are present in the dataframe? </font>
-  <font color='red'>(3821, 26)</font>
-  <font color='red'>(3879, 28)</font>
-  <font color='red'>(3853, 28)</font>
-  <font color='red'>(3866, 26)</font>

In [4]:
# Write you code here
movies.isnull().sum()


color                         2
director_name                 0
num_critic_for_reviews        1
duration                      1
director_facebook_likes       0
actor_3_facebook_likes        6
actor_2_name                  1
actor_1_facebook_likes        0
gross                         0
genres                        0
actor_1_name                  0
movie_title                   0
num_voted_users               0
cast_total_facebook_likes     0
actor_3_name                  6
facenumber_in_poster          6
plot_keywords                30
movie_imdb_link               0
num_user_for_reviews          0
language                      4
country                       0
content_rating               48
budget                        0
title_year                    0
actor_2_facebook_likes        1
imdb_score                    0
aspect_ratio                 72
movie_facebook_likes          0
dtype: int64

#### <font color='red'>Question 2: How many columns have null values present in them? Try writing a code for this instead of counting them manually.</font>

-  <font color='red'>3</font>
-  <font color='red'>6</font>
-  <font color='red'>9</font>
-  <font color='red'>12</font>

In [5]:
# Write you code here



### Task 2: Cleaning the Data

**Subtask 2.1: Drop unecessary columns**

For this assignment, you will mostly be analyzing the movies with respect to the ratings, gross collection, popularity of movies, etc. So many of the columns in this dataframe are not required. So it is advised to drop the following columns.
-  `color`
-  `director_facebook_likes`
-  `actor_1_facebook_likes`
-  `actor_2_facebook_likes`
-  `actor_3_facebook_likes`
-  `actor_2_name`
-  `cast_total_facebook_likes`
-  `actor_3_name`
-  `duration`
-  `facenumber_in_poster`
-  `content_rating`
-  `country`
-  `movie_imdb_link`
-  `aspect_ratio`
-  `plot_keywords`

In [6]:
# Check the 'drop' function in the Pandas library - dataframe.drop(list_of_unnecessary_columns, axis = )
# Write your code for dropping the columns here. It is advised to keep inspecting the dataframe after each set of operations

movies.drop([ 'color',
    'director_facebook_likes',
    'actor_1_facebook_likes',
    'actor_2_facebook_likes',
    'actor_3_facebook_likes',
    'actor_2_name',
    'cast_total_facebook_likes',
    'actor_3_name',
    'duration',
    'facenumber_in_poster',
    'content_rating',
    'country',
    'movie_imdb_link',
    'aspect_ratio',
    'plot_keywords'] , axis = 1 , inplace=True)

movies

director_name  num_critic_for_reviews        gross  \
0         James Cameron                   723.0  760505847.0   
1        Gore Verbinski                   302.0  309404152.0   
2            Sam Mendes                   602.0  200074175.0   
3     Christopher Nolan                   813.0  448130642.0   
4        Andrew Stanton                   462.0   73058679.0   
...                 ...                     ...          ...   
3848      Shane Carruth                   143.0     424760.0   
3849   Neill Dela Llana                    35.0      70071.0   
3850   Robert Rodriguez                    56.0    2040920.0   
3851       Edward Burns                    14.0       4584.0   
3852           Jon Gunn                    43.0      85222.0   

                                   genres     actor_1_name  \
0         Action|Adventure|Fantasy|Sci-Fi      CCH Pounder   
1                Action|Adventure|Fantasy      Johnny Depp   
2               Action|Adventure|Thriller  Christoph Waltz   
3                         Action|Thriller        Tom Hardy   
4                 Action|Adventure|Sci-Fi     Daryl Sabara   
...                                   ...              ...   
3848                Drama|Sci-Fi|Thriller    Shane Carruth   
3849                             Thriller      Ian Gamazon   
3850  Action|Crime|Drama|Romance|Thriller  Carlos Gallardo   
3851                         Comedy|Drama      Kerry Bishé   
3852                          Documentary      John August   

                                    movie_title  num_voted_users  \
0                                       Avatar            886204   
1     Pirates of the Caribbean: At World's End            471220   
2                                      Spectre            275868   
3                        The Dark Knight Rises           1144337   
4                                  John Carter            212204   
...                                         ...              ...   
3848                                    Primer             72639   
3849                                    Cavite               589   
3850                               El Mariachi             52055   
3851                                 Newlyweds              1338   
3852                         My Date with Drew              4285   

      num_user_for_reviews language       budget  title_year  imdb_score  \
0                   3054.0  English  237000000.0      2009.0         7.9   
1                   1238.0  English  300000000.0      2007.0         7.1   
2                    994.0  English  245000000.0      2015.0         6.8   
3                   2701.0  English  250000000.0      2012.0         8.5   
4                    738.0  English  263700000.0      2012.0         6.6   
...                    ...      ...          ...         ...         ...   
3848                 371.0  English       7000.0      2004.0         7.0   
3849                  35.0  English       7000.0      2005.0         6.3   
3850                 130.0  Spanish       7000.0      1992.0         6.9   
3851                  14.0  English       9000.0      2011.0         6.4   
3852                  84.0  English       1100.0      2004.0         6.6   

      movie_facebook_likes  
0                    33000  
1                        0  
2                    85000  
3                   164000  
4                    24000  
...                    ...  
3848                 19000  
3849                    74  
3850                     0  
3851                   413  
3852                   456  

[3853 rows x 13 columns]

#### <font color='red'>Question 3: What is the count of columns in the new dataframe? </font>
-  <font color='red'>10</font>
-  <font color='red'>13</font>
-  <font color='red'>15</font>
-  <font color='red'>17</font>

movies.count('columns')

**Subtask 2.2: Inspect Null values**

As you have seen above, there are null values in multiple columns of the dataframe 'movies'. Find out the percentage of null values in each column of the dataframe 'movies'. 

In [13]:
# Write you code here
# ((movies.isnull().sum() / len(movies) ) * 100).round(2)
round(100*(movies.isnull().sum()/len(movies.index)), 2)

director_name             0.00
num_critic_for_reviews    0.03
gross                     0.00
genres                    0.00
actor_1_name              0.00
movie_title               0.00
num_voted_users           0.00
num_user_for_reviews      0.00
language                  0.10
budget                    0.00
title_year                0.00
imdb_score                0.00
movie_facebook_likes      0.00
dtype: float64

In [14]:
movies.isnull().sum()

director_name             0
num_critic_for_reviews    1
gross                     0
genres                    0
actor_1_name              0
movie_title               0
num_voted_users           0
num_user_for_reviews      0
language                  4
budget                    0
title_year                0
imdb_score                0
movie_facebook_likes      0
dtype: int64

#### <font color='red'>Question 4: Which column has the highest percentage of null values? </font>
-  <font color='red'>language</font>
-  <font color='red'>genres</font>
-  <font color='red'>num_critic_for_reviews</font>
-  <font color='red'>imdb_score</font>

**Subtask 2.3: Fill NaN values**

You might notice that the `language` column has some NaN values. Here, on inspection, you will see that it is safe to replace all the missing values with `'English'`.

In [29]:
# Write your code for filling the NaN values in the 'language' column here
movies['language'].fillna('English' , inplace = True) 
count = len(movies[movies['language'] == 'English'])
count

3675

#### <font color='red'>Question 5: What is the count of movies made in English language after replacing the NaN values with English? </font>
-  <font color='red'>3670</font>
-  <font color='red'>3674</font>
-  <font color='red'>3668</font>
-  <font color='red'>3672</font>

### Task 3: Data Analysis

**Subtask 3.1: Change the unit of columns**

Convert the unit of the `budget` and `gross` columns from `$` to `million $`.

In [30]:
# Write your code for unit conversion here
movies['budget'] = movies['budget'] / 10000
movies['gross'] = movies['gross'] / 10000

movies

director_name  num_critic_for_reviews         gross  \
0         James Cameron                   723.0  7.605058e-26   
1        Gore Verbinski                   302.0  3.094042e-26   
2            Sam Mendes                   602.0  2.000742e-26   
3     Christopher Nolan                   813.0  4.481306e-26   
4        Andrew Stanton                   462.0  7.305868e-27   
...                 ...                     ...           ...   
3848      Shane Carruth                   143.0  4.247600e-29   
3849   Neill Dela Llana                    35.0  7.007100e-30   
3850   Robert Rodriguez                    56.0  2.040920e-28   
3851       Edward Burns                    14.0  4.584000e-31   
3852           Jon Gunn                    43.0  8.522200e-30   

                                   genres     actor_1_name  \
0         Action|Adventure|Fantasy|Sci-Fi      CCH Pounder   
1                Action|Adventure|Fantasy      Johnny Depp   
2               Action|Adventure|Thriller  Christoph Waltz   
3                         Action|Thriller        Tom Hardy   
4                 Action|Adventure|Sci-Fi     Daryl Sabara   
...                                   ...              ...   
3848                Drama|Sci-Fi|Thriller    Shane Carruth   
3849                             Thriller      Ian Gamazon   
3850  Action|Crime|Drama|Romance|Thriller  Carlos Gallardo   
3851                         Comedy|Drama      Kerry Bishé   
3852                          Documentary      John August   

                                    movie_title  num_voted_users  \
0                                       Avatar            886204   
1     Pirates of the Caribbean: At World's End            471220   
2                                      Spectre            275868   
3                        The Dark Knight Rises           1144337   
4                                  John Carter            212204   
...                                         ...              ...   
3848                                    Primer             72639   
3849                                    Cavite               589   
3850                               El Mariachi             52055   
3851                                 Newlyweds              1338   
3852                         My Date with Drew              4285   

      num_user_for_reviews language        budget  title_year  imdb_score  \
0                   3054.0  English  2.370000e-26      2009.0         7.9   
1                   1238.0  English  3.000000e-26      2007.0         7.1   
2                    994.0  English  2.450000e-26      2015.0         6.8   
3                   2701.0  English  2.500000e-26      2012.0         8.5   
4                    738.0  English  2.637000e-26      2012.0         6.6   
...                    ...      ...           ...         ...         ...   
3848                 371.0  English  7.000000e-31      2004.0         7.0   
3849                  35.0  English  7.000000e-31      2005.0         6.3   
3850                 130.0  Spanish  7.000000e-31      1992.0         6.9   
3851                  14.0  English  9.000000e-31      2011.0         6.4   
3852                  84.0  English  1.100000e-31      2004.0         6.6   

      movie_facebook_likes        profit  
0                    33000  5.235058e-04  
1                        0  9.404152e-06  
2                    85000 -4.492583e-05  
3                   164000  1.981306e-04  
4                    24000 -1.906413e-04  
...                    ...           ...  
3848                 19000  4.177600e-07  
3849                    74  6.307100e-08  
3850                     0  2.033920e-06  
3851                   413 -4.416000e-09  
3852                   456  8.412200e-08  

[3853 rows x 14 columns]

**Subtask 3.2: Find the movies with highest profit**

   1. Create a new column called `profit` which contains the difference of the two columns: `gross` and `budget`.
   2. Sort the dataframe using the `profit` column as reference. (Find which command can be used here to sort entries from the documentation)
   3. Extract the top ten profiting movies in descending order and store them in a new dataframe - `top10`

In [22]:
# Write your code for creating the profit column here
movies['profit'] = movies['gross'] - movies['budget']
movies

director_name  num_critic_for_reviews         gross  \
0         James Cameron                   723.0  7.605058e-04   
1        Gore Verbinski                   302.0  3.094042e-04   
2            Sam Mendes                   602.0  2.000742e-04   
3     Christopher Nolan                   813.0  4.481306e-04   
4        Andrew Stanton                   462.0  7.305868e-05   
...                 ...                     ...           ...   
3848      Shane Carruth                   143.0  4.247600e-07   
3849   Neill Dela Llana                    35.0  7.007100e-08   
3850   Robert Rodriguez                    56.0  2.040920e-06   
3851       Edward Burns                    14.0  4.584000e-09   
3852           Jon Gunn                    43.0  8.522200e-08   

                                   genres     actor_1_name  \
0         Action|Adventure|Fantasy|Sci-Fi      CCH Pounder   
1                Action|Adventure|Fantasy      Johnny Depp   
2               Action|Adventure|Thriller  Christoph Waltz   
3                         Action|Thriller        Tom Hardy   
4                 Action|Adventure|Sci-Fi     Daryl Sabara   
...                                   ...              ...   
3848                Drama|Sci-Fi|Thriller    Shane Carruth   
3849                             Thriller      Ian Gamazon   
3850  Action|Crime|Drama|Romance|Thriller  Carlos Gallardo   
3851                         Comedy|Drama      Kerry Bishé   
3852                          Documentary      John August   

                                    movie_title  num_voted_users  \
0                                       Avatar            886204   
1     Pirates of the Caribbean: At World's End            471220   
2                                      Spectre            275868   
3                        The Dark Knight Rises           1144337   
4                                  John Carter            212204   
...                                         ...              ...   
3848                                    Primer             72639   
3849                                    Cavite               589   
3850                               El Mariachi             52055   
3851                                 Newlyweds              1338   
3852                         My Date with Drew              4285   

      num_user_for_reviews language        budget  title_year  imdb_score  \
0                   3054.0  English  2.370000e-04      2009.0         7.9   
1                   1238.0  English  3.000000e-04      2007.0         7.1   
2                    994.0  English  2.450000e-04      2015.0         6.8   
3                   2701.0  English  2.500000e-04      2012.0         8.5   
4                    738.0  English  2.637000e-04      2012.0         6.6   
...                    ...      ...           ...         ...         ...   
3848                 371.0  English  7.000000e-09      2004.0         7.0   
3849                  35.0  English  7.000000e-09      2005.0         6.3   
3850                 130.0  Spanish  7.000000e-09      1992.0         6.9   
3851                  14.0  English  9.000000e-09      2011.0         6.4   
3852                  84.0  English  1.100000e-09      2004.0         6.6   

      movie_facebook_likes        profit  
0                    33000  5.235058e-04  
1                        0  9.404152e-06  
2                    85000 -4.492583e-05  
3                   164000  1.981306e-04  
4                    24000 -1.906413e-04  
...                    ...           ...  
3848                 19000  4.177600e-07  
3849                    74  6.307100e-08  
3850                     0  2.033920e-06  
3851                   413 -4.416000e-09  
3852                   456  8.412200e-08  

[3853 rows x 14 columns]

In [31]:
# Write your code for sorting the dataframe here
movies.sort_values(by='profit')

director_name  num_critic_for_reviews         gross  \
2678      Joon-ho Bong                   363.0  2.201412e-28   
3282    Chan-wook Park                   202.0  2.116670e-29   
2693      Lajos Koltai                    73.0  1.958880e-29   
2136    Hayao Miyazaki                   174.0  2.298191e-28   
2147   Katsuhiro Ôtomo                   105.0  4.103880e-29   
...                ...                     ...           ...   
2748  Steven Spielberg                   215.0  4.349495e-26   
2704      George Lucas                   282.0  4.609357e-26   
25       James Cameron                   315.0  6.586723e-26   
28     Colin Trevorrow                   644.0  6.521773e-26   
0        James Cameron                   723.0  7.605058e-26   

                                                 genres         actor_1_name  \
2678                         Comedy|Drama|Horror|Sci-Fi            Doona Bae   
3282                                        Crime|Drama         Min-sik Choi   
2693                                  Drama|Romance|War         Marcell Nagy   
2136                        Adventure|Animation|Fantasy        Minnie Driver   
2147  Action|Adventure|Animation|Family|Sci-Fi|Thriller     William Hootkins   
...                                                 ...                  ...   
2748                                      Family|Sci-Fi         Henry Thomas   
2704                    Action|Adventure|Fantasy|Sci-Fi        Harrison Ford   
25                                        Drama|Romance    Leonardo DiCaprio   
28                     Action|Adventure|Sci-Fi|Thriller  Bryce Dallas Howard   
0                       Action|Adventure|Fantasy|Sci-Fi          CCH Pounder   

                              movie_title  num_voted_users  \
2678                            The Host             68883   
3282                      Lady Vengeance             53508   
2693                            Fateless              5603   
2136                   Princess Mononoke            221552   
2147                            Steamboy             13727   
...                                   ...              ...   
2748          E.T. the Extra-Terrestrial            281842   
2704  Star Wars: Episode IV - A New Hope            911097   
25                               Titanic            793059   
28                        Jurassic World            418214   
0                                 Avatar            886204   

      num_user_for_reviews   language        budget  title_year  imdb_score  \
2678                 279.0     Korean  1.221550e-24      2006.0         7.0   
3282                 131.0     Korean  4.200000e-25      2005.0         7.7   
2693                  45.0  Hungarian  2.500000e-25      2005.0         7.1   
2136                 570.0   Japanese  2.400000e-25      1997.0         8.4   
2147                  79.0   Japanese  2.127520e-25      2004.0         6.9   
...                    ...        ...           ...         ...         ...   
2748                 515.0    English  1.050000e-27      1982.0         7.9   
2704                1470.0    English  1.100000e-27      1977.0         8.7   
25                  2528.0    English  2.000000e-26      1997.0         7.7   
28                  1290.0    English  1.500000e-26      2015.0         7.0   
0                   3054.0    English  2.370000e-26      2009.0         7.9   

      movie_facebook_likes    profit  
2678                  7000 -0.012213  
3282                  4000 -0.004200  
2693                   607 -0.002500  
2136                 11000 -0.002398  
2147                   973 -0.002127  
...                    ...       ...  
2748                 34000  0.000424  
2704                 33000  0.000450  
25                   26000  0.000459  
28                  150000  0.000502  
0                    33000  0.000524  

[3853 rows x 14 columns]

In [34]:
 # Write your code to get the top 10 profiting movies here
top10 = movies.nlargest(10, 'profit')
print(top10[['movie_title', 'profit']])

                                     movie_title    profit
0                                        Avatar   0.000524
28                               Jurassic World   0.000502
25                                      Titanic   0.000459
2704         Star Wars: Episode IV - A New Hope   0.000450
2748                 E.T. the Extra-Terrestrial   0.000424
16                                 The Avengers   0.000403
482                               The Lion King   0.000378
230   Star Wars: Episode I - The Phantom Menace   0.000360
64                              The Dark Knight   0.000348
419                            The Hunger Games   0.000330


**Checkpoint:** You might spot two movies directed by `James Cameron` in the list.

#### <font color='red'>Question 6: Which movie is ranked 5th from the top in the list obtained? </font>
-  <font color='red'>E.T. the Extra-Terrestrial</font>
-  <font color='red'>The Avengers</font>
-  <font color='red'>The Dark Knight</font>
-  <font color='red'>Titanic</font>

**Subtask 3.3: Find IMDb Top 250**

Create a new dataframe `IMDb_Top_250` and store the top 250 movies with the highest IMDb Rating (corresponding to the column: `imdb_score`). Also make sure that for all of these movies, the `num_voted_users` is greater than 25,000. 

Also add a `Rank` column containing the values 1 to 250 indicating the ranks of the corresponding films.

In [37]:
# Write your code for extracting the top 250 movies as per the IMDb score here. Make sure that you store it in a new dataframe 
# and name that dataframe as 'IMDb_Top_250'
filtered_movies = movies[movies['num_voted_users'] > 25000]
filtered_movies = filtered_movies.sort_values(by='imdb_score', ascending=False)
filtered_movies.head(250)

director_name  num_critic_for_reviews         gross  \
1795        Frank Darabont                   199.0  2.834147e-27   
3016  Francis Ford Coppola                   208.0  1.348220e-26   
64       Christopher Nolan                   645.0  5.333161e-26   
2543  Francis Ford Coppola                   149.0  5.730000e-27   
325          Peter Jackson                   328.0  3.770193e-26   
...                    ...                     ...           ...   
67             Jon Favreau                   486.0  3.182982e-26   
73              Doug Liman                   585.0  1.001895e-26   
3052           Yash Chopra                    29.0  2.921738e-28   
95           Peter Jackson                   645.0  3.030012e-26   
3170  Christophe Barratier                   112.0  3.629758e-28   

                              genres           actor_1_name  \
1795                     Crime|Drama         Morgan Freeman   
3016                     Crime|Drama              Al Pacino   
64       Action|Crime|Drama|Thriller         Christian Bale   
2543                     Crime|Drama         Robert De Niro   
325   Action|Adventure|Drama|Fantasy          Orlando Bloom   
...                              ...                    ...   
67           Action|Adventure|Sci-Fi      Robert Downey Jr.   
73           Action|Adventure|Sci-Fi             Tom Cruise   
3052           Drama|Musical|Romance         Shah Rukh Khan   
95                 Adventure|Fantasy           Aidan Turner   
3170                     Drama|Music  Jean-Baptiste Maunier   

                                         movie_title  num_voted_users  \
1795                       The Shawshank Redemption           1689764   
3016                                  The Godfather           1155770   
64                                  The Dark Knight           1676169   
2543                         The Godfather: Part II            790926   
325   The Lord of the Rings: The Return of the King           1215718   
...                                              ...              ...   
67                                         Iron Man            696338   
73                                 Edge of Tomorrow            431620   
3052                                     Veer-Zaara             34449   
95                The Hobbit: An Unexpected Journey            637246   
3170                                     The Chorus             44151   

      num_user_for_reviews language        budget  title_year  imdb_score  \
1795                4144.0  English  2.500000e-27      1994.0         9.3   
3016                2238.0  English  6.000000e-28      1972.0         9.2   
64                  4667.0  English  1.850000e-26      2008.0         9.0   
2543                 650.0  English  1.300000e-27      1974.0         9.0   
325                 3189.0  English  9.400000e-27      2003.0         8.9   
...                    ...      ...           ...         ...         ...   
67                  1055.0  English  1.400000e-26      2008.0         7.9   
73                   741.0  English  1.780000e-26      2014.0         7.9   
3052                 119.0    Hindi  7.000000e-28      2004.0         7.9   
95                  1367.0  English  1.800000e-26      2012.0         7.9   
3170                 110.0   French  5.500000e-28      2004.0         7.9   

      movie_facebook_likes    profit  
1795                108000  0.000003  
3016                 43000  0.000129  
64                   37000  0.000348  
2543                 14000  0.000044  
325                  16000  0.000283  
...                    ...       ...  
67                   10000  0.000178  
73                   77000 -0.000078  
3052                  2000 -0.000004  
95                  166000  0.000123  
3170                     0 -0.000002  

[250 rows x 14 columns]

#### <font color='red'>Question 7: Suppose movies are divided into 5 buckets based on the IMDb ratings: </font>
-  <font color='red'>7.5 to 8</font>
-  <font color='red'>8 to 8.5</font>
-  <font color='red'>8.5 to 9</font>
-  <font color='red'>9 to 9.5</font>
-  <font color='red'>9.5 to 10</font>

<font color = 'red'> Which bucket holds the maximum number of movies from *IMDb_Top_250*? </font>

**Subtask 3.4: Find the critic-favorite and audience-favorite actors**

   1. Create three new dataframes namely, `Meryl_Streep`, `Leo_Caprio`, and `Brad_Pitt` which contain the movies in which the actors: 'Meryl Streep', 'Leonardo DiCaprio', and 'Brad Pitt' are the lead actors. Use only the `actor_1_name` column for extraction. Also, make sure that you use the names 'Meryl Streep', 'Leonardo DiCaprio', and 'Brad Pitt' for the said extraction.
   2. Append the rows of all these dataframes and store them in a new dataframe named `Combined`.
   3. Group the combined dataframe using the `actor_1_name` column.
   4. Find the mean of the `num_critic_for_reviews` and `num_user_for_review` and identify the actors which have the highest mean.

In [ ]:
# Write your code for creating three new dataframes here
Meryl_Streep = # Include all movies in which Meryl_Streep is the lead

In [ ]:
Leo_Caprio = # Include all movies in which Leo_Caprio is the lead

In [ ]:
Brad_Pitt = # Include all movies in which Brad_Pitt is the lead

In [ ]:
# Write your code for combining the three dataframes here
Combined = 

In [ ]:
# Write your code for grouping the combined dataframe here


In [ ]:
# Write the code for finding the mean of critic reviews and audience reviews here


#### <font color='red'>Question 8: Which actor is highest rated among the three actors according to the user reviews? </font>
-  <font color='red'>Meryl Streep</font>
-  <font color='red'>Leonardo DiCaprio</font>
-  <font color='red'>Brad Pitt</font>

#### <font color='red'>Question 9: Which actor is highest rated among the three actors according to the critics?</font>
-  <font color='red'>Meryl Streep</font>
-  <font color='red'>Leonardo DiCaprio</font>
-  <font color='red'>Brad Pitt</font>